utils/python/CIME/XML/entry_id.py

```diff
@@ -157,12 +157,18 @@ def _set_value(self, node, value, vid=None, subgroup=None, ignore_type=False):
         subgroup is ignored in the general routine and applied in specific methods
         """
         expect(subgroup is None, "Subgroup not supported")
+        valid_values = self.get_optional_node("valid_values", root=node)
         if ignore_type:
             expect(type(value) is str, "Value must be type string if ignore_type is true")
-            node.set("value",value)
+            str_value = value
         else:
             type_str = self._get_type_info(node)
-            node.set("value", convert_to_string(value, type_str, vid))
+            str_value = convert_to_string(value, type_str, vid)
+        if valid_values is not None and valid_values.text is not None and not str_value.startswith('$'):
```

Comment:
The concern you raise is valid however I don't think that I can address it in this PR - I am aware of at least one case in which changing that warning to an error would break things - that is on mira/cetus we pass $COBALT_PARTNAME to runjob, this variable is not available on the submit node but is resolved on the run node when the job starts, we also pass a $LOCARGS which may or may not be defined in the environment for a given job.   I don't know how to apply the strictness that your comment implies without breaking this functionality.